WebDriverAgentLib/Categories/XCUIElement+FBUID.m

```diff
@@ -20,12 +21,19 @@ - (NSString *)fb_uid
   if ([self respondsToSelector:@selector(accessibilityElement)]) {
     return [FBElementUtils uidWithAccessibilityElement:[self performSelector:@selector(accessibilityElement)]];
   }
-  // With Xcode 10, using fb_lastSnapshot is faster than resolving and using the lastSnapshot property
+  // With Xcode 10, uniqueMatchingSnapshotWithError is not supported
   if (isSDKVersionLessThan(@"13.0")) {
     return self.fb_lastSnapshot.fb_uid;
   }
-  [self fb_nativeResolve];
-  return self.lastSnapshot.fb_uid;
+  NSError *error = nil;
+  // Using the Xcode 11 snapshot function used for resolving an element to validate existance and retrieve UID with the same snapshot
+  // Removes the need to take two snapshots (one for existance and one for resolving)
+  XCElementSnapshot *snapshot = [[self query] uniqueMatchingSnapshotWithError:&error];
```

Comment:
I'd rather check if query responds to `uniqueMatchingSnapshotWithError` and avoid checking the SDK version. also, we use our own `fb_query` wrapper.

Reply:
I wasn't sure about using `includingNonModalElements` here, that's the main reason I didn't use `fb_query`. Also - I went in line with the `fb_lastSnapshot` which also does not use the wrapper.

Reply:
ok, makes sense